eth/catalyst/api.go

```diff
@@ -867,15 +862,11 @@ func (api *ConsensusAPI) newPayload(params engine.ExecutableData, versionedHashe
 	// Hence, we use a lock here, to be sure that the previous call has finished before we
 	// check whether we already have the block locally.
 
-	// OP-Stack diff: payload must have empty extraData before Holocene and hold eip-1559 params after Holocene.
-	if cfg := api.eth.BlockChain().Config(); cfg.IsHolocene(params.Timestamp) {
-		if err := eip1559.ValidateHoloceneExtraData(params.ExtraData); err != nil {
+	// OP-Stack diff payload validation:
+	if cfg := api.eth.BlockChain().Config(); cfg.IsOptimism() {
+		if err := checkOptimismPayload(params, cfg); err != nil {
 			return api.invalid(err, nil), nil
```

Comment:
Here we are returning a `nil` error with an invalid status. Above in `NewPayloadV4` on L664, the upstream code returns a non-nil error and an invalid status. We should file an issue to audit our overall approach to this, keeping in mind how the CL client (e.g. op-node) will consume these values.

Reply:
https://github.com/ethereum-optimism/op-geth/issues/596